spec/controllers/assignments_controller_spec.rb

```diff
@@ -166,6 +170,19 @@
           .to redirect_to roster_assignments_path(roster, date: Time.zone.today)
       end
 
+      context 'when the end date is before the start date' do
+        before { params[:end_date] = Date.yesterday.to_fs(:db) }
```

Comment:
It's generally bad practice to "mutate" test setup data in `before` blocks - it's a pattern that can quickly get out of hand and produce some quirky behavior down the line.

Instead, what would be best for this example is for you to memoize `end_date` individually, much like `user_ids` and `starting_user_id` above.